src/demo-app/sticky-header/sticky-header-demo.scss

```diff
@@ -0,0 +1,80 @@
+.demo-sticky {
+  .demo-centered {
+    text-align: left;
+    -webkit-appearance: none;
+    -moz-appearance: none;
+    height: 500px;
+    overflow: auto;}
+
+  button {
+    position: fixed;
+    margin: 16px;
+    z-index: 99;
+  }
+
+  * {
+    margin: 0;
+    padding: 0;
+  }
+
+  body {
+    font: 14px/1.5 Helvetica, Arial, sans-serif;
+    -webkit-appearance: none;
+    -moz-appearance: none;
+
+  }
+
+  h1 {
+    font-size: 20px;
+    padding: 20px;
+    -webkit-appearance: none;
+    -moz-appearance: none;
+  }
+
+  h2 {
+    font-size: 20px;
+    padding: 20px;
+    -webkit-appearance: none;
+    -moz-appearance: none;
+  }
+
+  p {
+    text-indent: 4em;
+    margin-bottom: 1.5em;
```

Comment:
As discussed, this `margin-bottom` is what's causing the space between the headers. You can use `:not(:last-child)` to avoid putting the margin on the last item in the list to prevent it.